wolfcrypt/src/port/iotsafe/iotsafe.c

```diff
@@ -719,6 +719,79 @@ static int iotsafe_put_public_key(byte *pubkey_id, unsigned long id_size,
     }
     return ret;
 }
+#ifdef HAVE_HKDF
+static int iotsafe_hkdf_extract(byte* prk, const byte* salt, word32 saltLen,
+       byte* ikm, word32 ikmLen, int digest)
+{
+    int ret;
+    char *resp;
+	uint16_t hash_algo  = 0;
+    int len;
+
+    WOLFSSL_MSG("Enter iotsafe_hkdf_extract");
+     switch (digest) {
+        case WC_SHA256:
+         hash_algo = (uint16_t)1;
+         if (ikmLen == 0) {
+             len = WC_SHA256_DIGEST_SIZE;
+         }
+            break;
+        case WC_SHA384:
+          hash_algo = (uint16_t)2;
+           if (ikmLen == 0) {
+             len = WC_SHA384_DIGEST_SIZE;
+         }
+             break;
+        case WC_SHA512:
+            hash_algo = (uint16_t)4;
+             if (ikmLen == 0) {
+             len = WC_SHA512_DIGEST_SIZE;
+         }
+            break;
+        default:
+            break;
```

Comment:
this default should return `BAD_FUNC_ARG`

Reply:
also updated.